Unexpected warning output

**Describe the bug**
liboqs build should complete without error or warning output on supported platforms.

**To Reproduce**
run `cmake -GNinja -B build -S . && cd build && ninja`

**Expected behavior**
Correct build without error or warning output.

**Screenshots**
Actual output:
```
[1408/1442] Building C object tests/CMakeFiles/kat_sig_stfl.dir/kat_sig_stfl.c.o
/home/baentsch/git/oqs/liboqs/tests/kat_sig_stfl.c: In function ‘sig_stfl_kat’:
/home/baentsch/git/oqs/liboqs/tests/kat_sig_stfl.c:314:17: warning: ignoring return value of ‘fscanf’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  314 |                 fscanf(fp_rsp, "%llu", &sigs_remain);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/baentsch/git/oqs/liboqs/tests/kat_sig_stfl.c:323:17: warning: ignoring return value of ‘fscanf’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  323 |                 fscanf(fp_rsp, "%llu", &sigs_maximum);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[1442/1442] Linking C executable tests/dump_alg_info

```


**Environment (please complete the following information):**
 - OS: [Ubuntu 22]
 - OpenSSL version [3.0.2]
 - Compiler version used [gcc 11.4]
 - Build variables used [None -- see command above]
 - liboqs version [main branch]

Tagging @cothan @ashman-p as this might be related to missing STFL guards/#defines.

--- a/tests/kat_sig_stfl.c
+++ b/tests/kat_sig_stfl.c
@@ -311,7 +311,10 @@ OQS_STATUS sig_stfl_kat(const char *method_name, const char *katfile) {
 
 	// Echo back remain
 	if (FindMarker(fp_rsp, "remain = ")) {
-		fscanf(fp_rsp, "%llu", &sigs_remain);
+		if (EOF == fscanf(fp_rsp, "%llu", &sigs_remain)) {
+            fprintf(stderr, "[kat_stfl_sig] %s ERROR: unable to read 'remain' from <%s>\n", method_name, katfile);
+            goto err;
+        };
 		fprintf(fh, "remain = %llu\n", sigs_remain);
 	} else {
 		fprintf(stderr, "[kat_stfl_sig] %s ERROR: OQS_SIG_STFL_sigs_remaining failed!\n", method_name);
@@ -320,7 +323,10 @@ OQS_STATUS sig_stfl_kat(const char *method_name, const char *katfile) {
 
 	// Echo back max
 	if (FindMarker(fp_rsp, "max = ")) {
-		fscanf(fp_rsp, "%llu", &sigs_maximum);
+		if (EOF == fscanf(fp_rsp, "%llu", &sigs_maximum)) {
+            fprintf(stderr, "[kat_stfl_sig] %s ERROR: unable to read 'max' from <%s>\n", method_name, katfile);
+            goto err;
+        };
 		fprintf(fh, "max = %llu\n", sigs_maximum);
 	} else {
 		fprintf(stderr, "[kat_stfl_sig] %s ERROR: OQS_SIG_STFL_sigs_total failed!\n", method_name);

--- a/tests/kat_sig_stfl.c
+++ b/tests/kat_sig_stfl.c
@@ -312,9 +312,9 @@ OQS_STATUS sig_stfl_kat(const char *method_name, const char *katfile) {
 	// Echo back remain
 	if (FindMarker(fp_rsp, "remain = ")) {
 		if (EOF == fscanf(fp_rsp, "%llu", &sigs_remain)) {
-            fprintf(stderr, "[kat_stfl_sig] %s ERROR: unable to read 'remain' from <%s>\n", method_name, katfile);
-            goto err;
-        };
+			fprintf(stderr, "[kat_stfl_sig] %s ERROR: unable to read 'remain' from <%s>\n", method_name, katfile);
+			goto err;
+		};
 		fprintf(fh, "remain = %llu\n", sigs_remain);
 	} else {
 		fprintf(stderr, "[kat_stfl_sig] %s ERROR: OQS_SIG_STFL_sigs_remaining failed!\n", method_name);
@@ -324,9 +324,9 @@ OQS_STATUS sig_stfl_kat(const char *method_name, const char *katfile) {
 	// Echo back max
 	if (FindMarker(fp_rsp, "max = ")) {
 		if (EOF == fscanf(fp_rsp, "%llu", &sigs_maximum)) {
-            fprintf(stderr, "[kat_stfl_sig] %s ERROR: unable to read 'max' from <%s>\n", method_name, katfile);
-            goto err;
-        };
+			fprintf(stderr, "[kat_stfl_sig] %s ERROR: unable to read 'max' from <%s>\n", method_name, katfile);
+			goto err;
+		};
 		fprintf(fh, "max = %llu\n", sigs_maximum);
 	} else {
 		fprintf(stderr, "[kat_stfl_sig] %s ERROR: OQS_SIG_STFL_sigs_total failed!\n", method_name);

